Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -26,7 +26,7 @@ public IEnumerable<LookupViewModel> GetSeriesLookup(string type)
         {
             if (type == "status")
             {
-                var res = _context.SeriesStatus
+                var res = _context.SeriesStatus.Where(query => query.ID != 0)
```

Comment:
note from above, I don't see how this is really reducing any lines of code. As well as the fact that I would need to have a function return a template IEnumerable object so that I could return an Enumerable SeriesStatus or ActivityStatus or Activity... so on.

Reply:
This is not necessarily about reducing lines of code. Cam's point was that you have this business rule, `__Status == 0 means deleted`, that is spread out in tens of different queries. So, if you ever decided to change that rule (say you have a new sentinel value that means deleted), you would have to make sure to edit every single one of those queries.

More realistically, from now on everyone who doesn't want to show deleted items needs to remember to filter out anything with a deleted status. That's especially error-prone because most of the time developers aren't consciously thinking about deleted items. So there's a high likelihood that they would forget to add this filter and then oops - deleted entities are being shown on the front end!

Extracting the business rule to an extension method solves the first problem. It doesn't exactly solve the second problem, but it does lessen the burden on developers because it's so much easier to remember `_context.X.WithoutDeleted()` then `_context.X.Where(x => x.XStatusID != 0)`.

That said, I'm not sure how realistic it is to make such an extension method. There are, as far as I'm aware, two ways to do it:
1. Use reflection, so that at runtime, if the type `X` contains the special `XStatusID` member, then we filter to `X.Where(x => X.XStatusID != 0)`, and if `X` doesn't have such a member, then we return `X` unchanged. This approach would completely work, but it has two drawbacks. Firstly, reflection has a not insignificant runtime cost. Secondly, adding reflection would make it impossible to translate the Enumerable into a SQL query, so the query would always have to be performed in memory in the C# application, which as you know is never ideal.
2. Update all the entities that can be "deleted" in this way to inherit from an `IDeletable` interface. The interface would define the `.WithoutDeleted()` function, but each entity would have to implement it. This is a much more palatable solution. It unfortunately doesn't actually solve the "only change it in one place" issue, but it does still make that safer because instead of searching for all the magic `Where`s in the code, you would only need to update all the implementations of `IDeletable`. It does handle the second and more realistic problem perfectly.

If it isn't obvious from my framing, I am in favor of the second solution. I won't tell you that you have to do that, but I do think it would make the code better.